lib/hooks.ts

```diff
@@ -75,3 +75,25 @@ export const useOrder = (orderId: number) => {
         error,
     };
 }
+
+export const useShippingAndProductsOrder = (orderId: number) => {
+    const { context } = useSession();
+    const params = new URLSearchParams({ context }).toString();
+    const shouldFetch = context && orderId !== undefined;
+
+    // Conditionally fetch product if it doesn't exist in the list (e.g. deep linking)
+    const { data, error: orderError } = useSWR<Order, ErrorProps>(shouldFetch ? [`/api/orders/${orderId}`, params] : null, fetcher);
```

Comment:
Why don't we be extensible and use `useOrder` here?
```suggestion
    const { order, error: orderError } = useOrder(orderId);
```